### DISCUSSION TITLE
inline icons (e.g font-awesome, material design icons, octicons)

### DISCUSSION THREAD

**Discussion by JPRBM** [outline/outline#6283] [Ideas]
I'm coming over from 'Material for mkdocs ' and a feature I'm missing is using inline icons. We create a lot of application / process instructions for users in applications. A lot of modern web-application have chosen to use icons for buttons. As an easy reference we like to show the icon the user needs to press/search in our instructions.

example (icons in dutch text):
![image](https://github.com/outline/outline/assets/110032450/1d372a87-c0f3-4d6d-b8a1-63186f2e7e23)

Unfortunalty I'm not seeing any option to do this a the moment in outline. Even a small image won't show between text. It's always on the next line.

a direct link to the feature: https://squidfunk.github.io/mkdocs-material/reference/icons-emojis/#using-icons 
icons sets they support:
- [Material Design](https://materialdesignicons.com/)
- [FontAwesome](https://fontawesome.com/search?m=free)
- [Octicons](https://octicons.github.com/)
- [Simple Icons](https://simpleicons.org/)

It would be great in outline if the current emoticon function would be extended and typing ':fa-' would start listing font-awesome icons like :fa-solid-file and this insert the icon inline (for example)

**Comment by tommoor** [comment]
I think the way to achieve this would be to allow adding custom emoji upload to Outline, and then you would be able to insert the custom icons in the same way as any other emoji 🤔

**Reply by tmeuze** [reply]
Would you please consider adding the [Material Design Icons (MDI)](https://github.com/Templarian/MaterialDesign?tab=readme-ov-file) [library](https://pictogrammers.com/docs/guides/webfont-alternatives/), and having the icon picker appear after typing `:` :

![CleanShot 2024-12-02 at 18 52 30@2x](https://github.com/user-attachments/assets/358988b7-8e1c-49b9-829d-a0fe44ede96f)

**Reply by tmeuze** [reply]
I see that FontAwesome might be in the pipeline, and I'm thrilled. Thanks!

**Comment by AlexandreHT** [comment]
I have the exact same need with the difference that I write procedure of Commercial off-the-shelf software, so I don't have a base of custom icons. 
Each time the button to insert as an inline image needs to be copied/pasted from the screen. Therefore the custom emoji potential solution would probably not be practical/feasible.
